File: SignalRDashboard.Web/Scripts/app/models/demo/SiteStatus.js
(function(signalrdashboardDemo) {
    signalrdashboardDemo.SiteStatus = SiteStatus;

    function SiteStatus() {
        // Properties matching data model received via SignalR
        this.siteName = "";
        this.url = "";
        this.isSiteUp = true;

        // Error tracking properties
        this.hasSiteDownError = false;

        // Error check conditions
        var model = this;
        var errorChecks = [];
        errorChecks.push({ model: model, condition: function() { return !model.isSiteUp; }, targetProperty: 'hasSiteDownError' });

        this.updateFromData = function(data) {            
            signalrdashboard.mapping.map(data, this);
            signalrdashboard.alertHandler.performErrorChecks('DemoSiteStatus', errorChecks);
        };
    }
    
})(window.signalrdashboard.demo || (window.signalrdashboard.demo = {}));

File: SignalRDashboard.Data.Demo/Hubs/Models/SiteStatus.cs
using System;
using SignalRDashboard.Data.Core.Hubs.Models;

namespace SignalRDashboard.Data.Demo.Hubs.Models
{
    public class SiteStatus : DashboardHubModel
    {
        private string _siteName;
        public string SiteName
        {
            get { return _siteName; }
            set { SetProperty(ref _siteName, value); }
        }

        private bool _isSiteUp;
        public bool IsSiteUp
        {
            get { return _isSiteUp; }
            set
            {
                SetProperty(ref _isSiteUp, value);
                if (!_isSiteUp)
                {
                    LastDownTime = DateTime.Now.ToShortDateString();
                }
            }
        }

        private string _url;
        public string Url
        {
            get { return _url; }
            set { SetProperty(ref _url, value); }
        }

        private string _lastDownTime;
        public string LastDownTime
        {
            get { return _lastDownTime; }
            set { SetProperty(ref _lastDownTime, value); }
        }
    }
}
